pkg/stacks/unpack.go

```diff
@@ -219,12 +211,24 @@ func (sp *StackPackage) IsNamespaced() bool {
 // SetApp sets the Stack's App metadata
 func (sp *StackPackage) SetApp(app v1alpha1.AppMetadataSpec) {
 	sp.Stack.Spec.AppMetadataSpec = app
+	sp.TemplateStack.Spec.AppMetadataSpec = app
 	sp.appSet = true
 }
 
-// SetBehavior sets the Stack's Behavior configuration
-func (sp *StackPackage) SetBehavior(behavior unstructured.Unstructured) {
-	sp.Behavior = behavior
+// SetStack sets the Stack's configuration
+// This is primarily for defining Template Stack behaviors, but is intended
+// to replace SetApp in a merger of TemplateStack and Stack
+func (sp *StackPackage) SetStack(tss v1alpha1.TemplateStackSpec) {
+	tss.AppMetadataSpec.DeepCopyInto(&sp.Stack.Spec.AppMetadataSpec)
+
+	sp.TemplateStack.Spec.StackConfigurationSpec = tss.StackConfigurationSpec
+
+	deployment := appsv1.Deployment{}
```

Comment:
For generating this deployment, we'll need to inspect the template stack / stack definition to either:

* Pass the namespaced name of the stack definition into the controller, OR
* Pass the GVK of the CRD and the namespaced name of the stack definition into the controller

Also, just making a note that we were imagining setting an owner reference for the stack which points back to the stack definition (though I've got some code which does that in #1214)

Reply:
Closing this as resolved since this code has changed. Please resurface this concern if it is still applies.